1-foundations/2-control-flows-and-functions/foundations-s2.ipynb

```diff
@@ -726,18 +1268,264 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 136,
+   "id": "3870bea5-b2a7-4d69-b27c-ea849a377ddd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### YOR CODE HERE\n",
+    "\n",
+    "## Your code must run without AssertionError. If you get the error, something is not right\n",
+    "\n",
+    "\n",
+    "    \n",
+    "\n",
+    "\n",
+    "## ======= DO NOT EDIT THIS PART========\n",
+    "\n",
+    "x = max_3(10, 7, 28)\n",
+    "assert x == 28, \"This is not the max element\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "20bebd4f-16f0-4f41-84cf-9747d4ed60c4",
+   "metadata": {},
+   "source": [
+    "#### Task 6\n",
+    "\n",
+    "Write a function that takes a string as an input but returns the reverse of it. For instance, if you input \"abcde\", it must return \"edcba\". Name the function `string_reverse` that takes one argument. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 143,
+   "id": "276e9a13-f722-4348-b26c-3a0e5a06625c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### YOUR CODE HERE\n",
+    "\n",
+    "## HINT: Create a temporary variable. Loop through the elements of the string and store them in the temporary variable. \n",
+    "\n",
+    "\n",
+    "\n",
+    "## ======= DO NOT EDIT THIS PART========\n",
+    "\n",
+    "assert string_reverse(\"rcb1096281\") == \"1826901bcr\", \"Oops! Guess that didn't work. Look the hint provided and try again\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "57971df7-ca64-4240-81ac-e26a8297dae0",
+   "metadata": {},
+   "source": [
+    "#### Task 7\n",
+    "\n",
+    "Create a function that calculates GDP per capita for any given country. Name the function `gdp_percapita` and it should accept two arguments : country code (ISO3) and year. Either name the arguments country, series, time or keep the order consistent in order to avoid validation error. \n",
+    "\n",
+    "i.e. your function should look like gdp_percapita(country, time) or should also work with gdp_percapita('AFG', 2010).\n",
+    "\n",
+    "Use World Bank's Open Data API to get the data and compute GDP per capita. \n",
+    "\n",
+    "Logic for calculation (`GDP per capita (current US$)`) : `GDP (current US$)` / `Population (total)`\n",
+    "\n",
+    "Feel free to re-use the code in `query_population` function above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 148,
+   "id": "bc964d53-2076-47e9-b304-a258dba6b394",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### YOUR CODE HERE\n",
+    "\n",
+    "## HINT: You'll have to make two requests to get data for two variables GDP and Population. Series code for GDP is \"NY.GDP.MKTP.CD\" and for Population is \"NY.GDP.MKTP.CD\". \n",
+    "## You can take help from query_population function and substitute the series code with the one provided above. Index the request and get the values returned for that country and time period. \n",
```

Comment:
```suggestion
    "## HINT: You'll have to make two requests to get data for two variables GDP and Population. \n"
    "## Series code for GDP is \"NY.GDP.MKTP.CD\" and for Population is \"NY.GDP.MKTP.CD\". \n",
    "## You can take help from query_population function and substitute the series code with \n"
    "##  the one provided above. Index the request and get the values returned for that country and time period. \n",
```
Markdown wraps text but not comments. Breaking this up to make it readable without scrolling